File: app/src/main/java/com/techventus/wikipedianews/dialogfragment/BaseDialogFragment.java
package com.techventus.wikipedianews.dialogfragment;

import android.app.AlertDialog;
import android.graphics.Typeface;
import android.os.Bundle;
import android.widget.Button;

import androidx.core.content.ContextCompat;
import androidx.fragment.app.DialogFragment;

import com.techventus.wikipedianews.R;
import com.techventus.wikipedianews.WikiApplication;

public abstract class BaseDialogFragment extends DialogFragment {
	private static final Typeface mTypeface;
	static
	{
		mTypeface = Typeface.createFromAsset(WikiApplication.getInstance().getAssets(), "fonts/Calibre-Semibold.otf");
	}

	@Override
	public void onStart()
	{
		super.onStart();
		AlertDialog alertDialog = (AlertDialog) getDialog();
		if (alertDialog != null)
		{
			updateButton(alertDialog.getButton(AlertDialog.BUTTON_POSITIVE));
			updateButton(alertDialog.getButton(AlertDialog.BUTTON_NEGATIVE));
		}
	}

	private void updateButton(final Button button)
	{
		if (button != null && getActivity() != null)
		{
			button.setTypeface(mTypeface);
			button.setTextColor(ContextCompat.getColor(getActivity(), R.color.notification_dialog_button));
		}
	}
}

File: app/src/main/java/com/techventus/wikipedianews/dialogfragment/BaseDialogFragment.kt
package com.techventus.wikipedianews.dialogfragment

import android.app.AlertDialog
import android.graphics.Typeface
import android.os.Bundle
import android.widget.Button
import androidx.core.content.ContextCompat
import androidx.fragment.app.DialogFragment
import com.techventus.wikipedianews.R
import com.techventus.wikipedianews.WikiApplication

abstract class BaseDialogFragment : DialogFragment() {

    override fun onStart() {
        super.onStart()
        val alertDialog = dialog as AlertDialog?
        alertDialog?.let {
            updateButton(it.getButton(AlertDialog.BUTTON_POSITIVE))
            updateButton(it.getButton(AlertDialog.BUTTON_NEGATIVE))
        }
    }

    private fun updateButton(button: Button?) {
        button?.let {
            it.typeface = mTypeface
            activity?.let { currentActivity ->
                it.setTextColor(ContextCompat.getColor(currentActivity, R.color.notification_dialog_button))
            }
        }
    }

    companion object {
        private val mTypeface: Typeface = Typeface.createFromAsset(WikiApplication.getInstance()?.assets, "fonts/Calibre-Semibold.otf")
    }
}